flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/codegen/ValuesCodeGenerator.scala

```diff
@@ -43,9 +44,17 @@ object ValuesCodeGenerator {
     val generatedRecords = tuples.map {
       r =>
         exprGenerator.generateResultExpression(
-          r.map(exprGenerator.generateExpression),
+          r.zipWithIndex.map {
```

Comment:
Probably as a result of https://issues.apache.org/jira/browse/CALCITE-4383
it requires same type now... To achieve there is a cast to least restrictive